### PR TITLE
fix(DB/Smart_scripts) Razorfen Stalker

### LINKED ISSUE #4668: Razorfen Stalkers don't have nameplates [$5]
Razorfen Stalkers don't have nameplates

##### CURRENT BEHAVIOUR:

This NPC doesn't have nameplates during combat (press V).

https://classic.wowhead.com/npc=6035/razorfen-stalker

WoWScrnShot_030121_143844

##### EXPECTED BLIZZLIKE BEHAVIOUR:

I'd guess they're meant to have nameplates? It could be related to their "Prowl"/Stealth skill they have?

##### STEPS TO REPRODUCE THE PROBLEM:

1. Go to Razorfen Kraul
2. Enable Nameplates (usuall V or Control-V)
3. Find and fight a Razorfen Stalker

##### AC HASH/COMMIT:

AzerothCore rev. c2f274e06dcf 2021-02-28 13:31:49 +0000 (master branch) (Unix, RelWithDebInfo)

##### OTHER CUSTOMIZATIONS:

<bountysource-plugin>

---
There is a **[$5 open bounty](https://www.bountysource.com/issues/96862296-razorfen-stalkers-don-t-have-nameplates?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Branel**
## Changes Proposed:
-  Adds Sneak to Razorfen Stalker
-  Updates Backstab comment

## Issues Addressed:
- Closes #4668

## SOURCE:

Original bug report.

## Tests Performed:

- Tested on Windows 10, tested by entering the dungeon and going to the creature.

## How to Test the Changes:

[Please follow this tutorial](https://www.azerothcore.org/wiki/How-to-test-DB-only-changes), if done correctly use `.go creature 87329`
Razorfen Stalker should now show his nameplate when out of stealth, and upon combat reset he will return to stealth.

## Known Issues and TODO List:

N/A

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1618834063609499000.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1618834063609499000');
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 6035;
+DELETE FROM `smart_scripts` WHERE (`entryorguid` = 6035) AND (`source_type` = 0) AND (`id` IN (0, 1));
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(6035, 0, 0, 0, 67, 0, 100, 0, 5000, 5000, 0, 0, 0, 11, 7159, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 'Razorfen Stalker - On Behind Target - Cast \'Backstab\''),
+(6035, 0, 1, 0, 1, 0, 100, 0, 0, 0, 1000, 1000, 0, 11, 22766, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Razorfen Stalker - Out of Combat - Cast \'Sneak\'');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1618834063609499000.sql:6]: newline

**Comment by Branel** [comment]
@Kitzunu Please let me know if its any better now, this might be a bit overkill.
Thanks for your feedback!

**Comment by Kitzunu** [comment]
What I mean with newline is that on the end of every file needs to be a blank.
In GitHub view you can see a red circle, which indicates there is no newline.

So for example, if your last line is L#15 you need to add an extra L#16 without data

**Comment by Branel** [comment]
That actually makes a lot of sense, thank you again.

**Review by Si1ker — APPROVED** [review]

**Comment by Si1ker** [comment]
Tested, it has the nameplate now.

image

**Comment by Si1ker** [comment]
Hey @Branel  thanks for the PR.

Do you know what could be the cause for this other mob with the same behavior?

The NPC with the ID 8881
.go c id 8881

**Comment by Branel** [comment]
When I first found this issue I investigated both but since there was more information about the Razorfen Stalker I first went for that one. As much as I learned from solving this there is something missing for this creature that makes the nameplate not want to appear. There are NPCs with the exact same model e.g `.go c 4154` that are completly fine. I tried looking at a few stuff but no avail. I will open a seperate bug report and see if someone has better luck, and possibly investigate further when I have more time.

**Comment by Si1ker** [comment]
> When I first found this issue I investigated both but since there was more information about the Razorfen Stalker I first went for that one. As much as I learned from solving this there is something missing for this creature that makes the nameplate not want to appear. There are NPCs with the exact same model e.g `.go c 4154` that are completly fine. I tried looking at a few stuff but no avail. If you could (I could aswell) open a seperate bug report and we'll see if someone has better luck, and possibly I could investigate further when I have more time.

@Branel sounds good, thanks.

**Comment by pklloveyou** [comment]
Tested

**Pull request merged**